vm/src/rv32/parse.rs

```diff
@@ -121,6 +121,8 @@ fn sop(word: u32) -> Option<SOP> {
 fn aop(word: u32) -> Option<AOP> {
```

Comment:
@yoichi-nexus It seems to me that we need to enforce `opcode(word) = 0b0110011` for all these arithmetic operations and return `None` when this is not the case. It seems to me that we also need to enforce `func7(word) = 0b0100000` for `SUB` and `SRA` and  `func7(word) = 0b0000000` for all the remaining arithmetic operations.

Reply:
Actually, line 215 seems to be already enforcing the check `opcode(word) = 0b0110011` so it's not clear to me that you need to check it again in the `aop` function. However, the check for `func7(word) = 0b0100000` for `SUB` and `SRA` and  `func7(word) = 0b0000000` for all the remaining arithmetic operations still seems to be needed.

Reply:
@michel-nexus Please note that the function `aop()` is called both for `ADD` and `ADDI`. `aop()` needs to look at the opcode again to see if it's about `ADD` or `ADDI`. In case of `ADD`, `aop()` needs to enforce `func7 == 0`. In case of `ADDI`, `aop()` doesn't need to enforce anything on `func7`.

Checks about `SUB`, `SRA`, `AND`, `OR`, `XOR` etc are still coming.

Reply:
I've applied the comments. Especially I separated `aop()` into two functions each dealing with the `ADD` family and `ADDI` family.